### PR TITLE
Requesting lots of memory for Prometheus

### PR DESCRIPTION
**Pull request by johnc1231**
No more OOMs for Prometheus.

### CODE REVIEW

**Comment by johnc1231** [comment]
The nodes only have 30Gi total. If I make it exactly 30 I imagine there will be issues since the nodes also have to run docker and logging stuff and whatnot, right?

**Comment by johnc1231** [comment]
Also, now that we are not going with the "run a highmem node premptible pool" approach, should we still have Prometheus tolerate preemptibles? It would probably be better if it was on one of our always running nodes, especially since it has a nontrivial amount of startup time in my experience

**Comment by danking** [comment]
Non preemptible for that reason seems fine. 

However, I think I have a fundamental misunderstanding. In my experience thus far,  Prometheus will need more than 30GB of RAM if anyone runs thousands of pods on our cluster for an hour or more. Is that not your understanding? Last time I ran the test Prometheus wasn’t able to start after crashing. Also the per-pod Graphana graphs loaded noticeably slower than everything else.

**Comment by johnc1231** [comment]
I think I had the misunderstanding. I know Prometheus dies when we do big tests, but I thought that was just because we requested no resources at all in the past. I didn't understand that it consistently uses 30Gb + of memory at those times. In that case, it seems constantly running a bigger node is unavoidable

**Comment by danking** [comment]
Agreed. We tend to idle at 5 n1-standard-8 non-preemptible machines. An additional pool of n1-standard-16's or 32's would be fine. Prometheus could take, say, 45 GB, and a bunch of compute-hungry but memory-lean tasks will get co-scheduled.

**Comment by danking** [comment]
Hrm. One issue: I do not fully understand the minimum number of nodes allowed in GKE. Clearly, we need at least three master nodes to reach consensus. I'm not sure if any similar requirements exist for worker pools.

**Comment by johnc1231** [comment]
> "A node pool can contain only a single node or many nodes"
https://cloud.google.com/kubernetes-engine/docs/concepts/node-pools

**Comment by johnc1231** [comment]
Final conclusion here: Since batch 1 is discontinued in favor of batch 2, and we don't know how prometheus will react to batch 2's logging load (should be better since there's not constant kubernetes junk), let's not start a new node pool for now, and request 20 Gbs of memory as done here.

File: monitoring/monitoring.yaml [hunk 1]
```diff
@@ -346,6 +346,10 @@ spec:
             name: etc-prometheus
           - mountPath: "/prometheus"
             name: prometheus-storage
+         resources:
+          requests:
```
Comment (danking) [inline monitoring/monitoring.yaml]: just realized your indentation is off.

Comment (johnc1231) [reply monitoring/monitoring.yaml]: Resolved

**Pull request merged**